### PR TITLE
Setup build pipeline with terraform

### PR DESCRIPTION
**Pull request by scotthain**
@chef/jex-team 

supersedes https://github.com/chef/chef-web-docs/pull/1364

### CODE REVIEW

File: .expeditor/buildkite/deploy_docs.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+set -eou pipefail
+
+aws-configure chef-cd
+
+ceval "aws s3 sync build s3://chef-web-docs-${acceptance}.cd.chef.co --profile chef-cd"
```
Comment (tduffield) [inline .expeditor/buildkite/deploy_docs.sh]: you aren't setting the variable `acceptance` - did you mean for this to just be a full string?

File: .expeditor/deploy.pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+steps:
+  - label: "Terraform"
+    command: .expeditor/buildkite/terraform.sh
+    concurrency: 1
```
Comment (tduffield) [inline .expeditor/deploy.pipeline.yml:4]: You need to specify a `concurrency_group` in both steps

File: Makefile [hunk 3]
```diff
@@ -2,12 +2,18 @@ BUILDDIR = build
 BUILD_COMMAND = sphinx-build -a -W
 BUILD_COMMAND_AND_ARGS = $(BUILD_COMMAND)
 
-docs:
+# if we are in CI we need to add python's binpath:
+export PATH := $(shell hab pkg path core/python2)/bin:$(PATH)
```
Comment (tduffield) [inline Makefile]: Can you clarify how/why this is only evaluated in CI and not locally?

File: terraform/s3.tf [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#########################################################################
+# Non production chef-web-docs s3 static site bucket
+#########################################################################
+
+module "chef-web-docs" {
+  source      = "git@github.com:chef/es-terraform.git//modules/cd_s3_website"
+  subdomain   = "chef-web-docs${var.environment == "delivered" ? "" : "-${var.environment}"}"
+  tag_contact = "shain"
```
Comment (tduffield) [inline terraform/s3.tf]: we probably want to make this `releng`

**Review by tduffield — CHANGES_REQUESTED** [review]
Just a few minor things, otherwise looks good!

File: .expeditor/deploy.pipeline.yml (hunk 2 above)
Comment (scotthain) [reply .expeditor/deploy.pipeline.yml:4]: is there a default or do I need to create one somewhere?

File: Makefile (hunk 3 above)
Comment (scotthain) [reply Makefile]: This is a misleading comment - we just need to add python to the path.

File: .expeditor/buildkite/deploy_docs.sh (hunk 1 above)
Comment (scotthain) [reply .expeditor/buildkite/deploy_docs.sh]: Copy paste error - should have been ENVIRONMENT

File: .expeditor/deploy.pipeline.yml (hunk 2 above)
Comment (tduffield) [reply .expeditor/deploy.pipeline.yml:4]: you just have to specify it like so: https://github.com/chef/a2/blob/master/.expeditor/deploy.pipeline.yml#L5

**Pull request closed without merging**